pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -131,8 +131,20 @@ func (esc *endpointSliceController) onEventInternal(_, ep *v1.EndpointSlice, eve
 			continue
 		}
 
+		// pure HTTP headless services should not need a full push since they do not
+		// require a Listener based on IP: https://github.com/istio/istio/issues/48207
+		pureHttp := false
+		for _, p := range modelSvc.Ports {
+			if p.Protocol.IsHTTP() {
+				pureHttp = true
+			} else {
+				pureHttp = false
+				break
+			}
+		}
+
 		esc.c.opts.XDSUpdater.ConfigUpdate(&model.PushRequest{
-			Full: features.EnableHeadlessService,
+			Full: features.EnableHeadlessService && !pureHttp,
```

Comment:
Do we need any push? I think pushEDS on L118 might do it?

Reply:
maybe? I'll test it out

Reply:
do you know what the `break` is doing on ln 153?  Iterating over a list of Services but only pushing an update for the first one?

Reply:
> happens to work because ConfigsUpdated is ignored for Full=true 

that was my thought too - but even then it's technically broken if `features.EnableHeadlessService` is false since it's potentially breaking early for non-full pushes